pyiceberg/catalog/glue.py

```diff
@@ -177,6 +191,23 @@ def _create_glue_table(self, database_name: str, table_name: str, table_input: T
         except self.glue.exceptions.EntityNotFoundException as e:
             raise NoSuchNamespaceError(f"Database {database_name} does not exist") from e
 
+    def _update_glue_table(self, database_name: str, table_name: str, table_input: TableInputTypeDef, version_id: str) -> None:
+        try:
+            self.glue.update_table(DatabaseName=database_name, TableInput=table_input, VersionId=version_id)
```

Comment:
every time that a glue table is updated there a a new version is created, and the previous versions are retained by default. The amount of table versions per AWS account is limited, and I've seen such limited reached many times specifically when using iceberg - see also this issue: https://github.com/dbt-athena/dbt-athena/issues/524 and this one https://github.com/dbt-athena/dbt-athena/pull/522

I'm wondering if you considered using `SkipArchive ` set to True by default?
Alternatively, you can give the final user control on such parameter. 

Previous table versions are only relevant for debugging e.g. spotting which was the old metadata location, but not really helpful for operations like snapshot rollback, where you need to use spark for it.

Reply:
Thanks for the great suggestion! Yes, I think we want to default `SkipArchive` to `TRUE` to avoid the same issue we saw in java: https://github.com/apache/iceberg/pull/6916

Reply:
Thanks, I saw so many workaround to prevent this issue, if such feature is already backed in will be amazing, as it avoids headaches.